libraries/AP_InertialSensor/AP_InertialSensor_SITL.cpp

```diff
@@ -122,17 +125,10 @@ void AP_InertialSensor_SITL::generate_accel()
         // VIB_FREQ is a static vibration applied to each axis
         const Vector3f &vibe_freq = sitl->vibe_freq;
 
-        if (vibe_freq.is_zero() && is_zero(sitl->vibe_motor)) {
-            // no rpm noise, so add in background noise if any
-            xAccel += accel_noise * rand_float();
-            yAccel += accel_noise * rand_float();
-            zAccel += accel_noise * rand_float();
-        }
-
         if (!vibe_freq.is_zero() && motors_on) {
-            xAccel += sinf(accel_time * 2 * M_PI * vibe_freq.x) * calculate_noise(accel_noise, noise_variation);
-            yAccel += sinf(accel_time * 2 * M_PI * vibe_freq.y) * calculate_noise(accel_noise, noise_variation);
-            zAccel += sinf(accel_time * 2 * M_PI * vibe_freq.z) * calculate_noise(accel_noise, noise_variation);
+            accel.x += sinf(accel_time * 2 * M_PI * vibe_freq.x) * calculate_noise(accel_noise, noise_variation);
+            accel.y += sinf(accel_time * 2 * M_PI * vibe_freq.y) * calculate_noise(accel_noise, noise_variation);
+            accel.z += sinf(accel_time * 2 * M_PI * vibe_freq.z) * calculate_noise(accel_noise, noise_variation);
```

Comment:
Not for this PR, but it would be nice to be able to:

```suggestion
            accel += sinf(accel_time * 2 * M_PI * vibe_freq) * calculate_noise_3f(accel_noise, noise_variation);
```